concordium-sdk/src/main/java/com/concordium/sdk/crypto/wallet/ConcordiumHdWallet.java

```diff
@@ -0,0 +1,328 @@
+        return fromSeedPhrase(split, network);
+    }
+
+    /**
+     * Create a Concordium hierarchical deterministic wallet from a seed phrase
+     * provided as a list of strings.
+     * @param seedPhrase the seed phrase with words separated by a single space
+     * @param network the network the wallet is for
+     * @return a wallet that can derive required Concordium wallet keys
+     * @throws IOException
+     * @throws MnemonicException
+     */
+    public static ConcordiumHdWallet fromSeedPhrase(List<String> seedPhrase, Network network) throws IOException, MnemonicException {
+        // Validate whether the input seed phrase is valid or not.
+        StringBuilder builder = new StringBuilder(WordLists.ENGLISH);
+        InputStream in = new ByteArrayInputStream(builder.toString().getBytes("UTF-8"));
+        String BIP39_ENGLISH_SHA256 = "ad90bf3beb7b0eb7e5acd74727dc0da96e0a280a258354e7293fb7e211ac03db";
+        MnemonicCode mnemonicCode = new MnemonicCode(in, BIP39_ENGLISH_SHA256);
+
+        // This throws an exception if the seed phrase is not valid.
+        mnemonicCode.check(seedPhrase);
+
+        byte[] seed = MnemonicCode.toSeed(seedPhrase, "");
+        StringBuilder sb = new StringBuilder();
+        for (byte b : seed) {
+            sb.append(String.format("%02X ", b));
+        }
+        return new ConcordiumHdWallet(sb.toString(), network);
+    }
+
+    /**
+     * Create a Concordium hierarchical deterministic wallet from a seed.
+     * @param seedAsHex the seed encoded as a hex string. Must be 128 characters.
+     * @param network the network the wallet is for.
+     * @return a wallet that can derive required Concordium wallet keys
+     */
+    public static ConcordiumHdWallet fromHex(String seedAsHex, Network network) {
+        if (seedAsHex.length() != 128) {
+            throw new IllegalArgumentException(
+                    "The provided seed " + seedAsHex + " is invalid as its length was not 128.");
+        } else if (!HexadecimalValidator.isHexadecimal(seedAsHex)) {
+            throw new IllegalArgumentException("The provided seed " + seedAsHex + " is not a valid hexadecimal string");
+        }
+
+        return new ConcordiumHdWallet(seedAsHex, network);
+    }
+
+    private enum UnsignedIntegerType {
+        U8,
+        U32,
+        U64;
+    }
+
+    private void checkUnsignedIntegerValues(UnsignedIntegerType type, long ...values) {
+        long maxValue;
+        switch (type) {
+            case U8: 
+                maxValue = 255;
+                break;
+            case U32:
+                maxValue = 4294967295l;
+                break;
+            case U64:
+                // Note that a long does not allow for a full u64 as it is a i64.
+                // The types in the SDK would need to be updated to fully accomodate a u64,
+                // and at that point this should also be changed.
+                maxValue = Long.MAX_VALUE;
```

Comment:
Why check for a max value here? If it fits in a long, then it's a 64 bit integer. 

What if the `long` was created with https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#parseUnsignedLong-java.lang.String- and a value `Long.MAX_VALUE +1`.

Reply:
The base problem is that there are places in the SDK that uses `long`, when it should be `UInt64`. I don't think that our changes here should be doing that breaking change, so instead I am just trying to protect us from getting bad values.

The check on `maxValue` is redundant when it is a `U64` but it was easier to do the redundant check than to handle it differently. I could refactor it to be handled by itself if you think that is more readable.

If you overflow then the value just becomes negative, and the check `value < 0` should catch that.

Reply:
We had an offline talk and the result of that meant I could remove all this.